### ISSUE TITLE
Multicast group joined and then left

### ISSUE THREAD

**Issue by fhunleth** [pcmarks/mdns_lite#66]
I haven't tracked this down, but it's weird so I figured that I'd post it. `224.0.0.251` is the mDNS multicast address. The network should send out an IGMP join message when `mdns_lite` starts listening on the address.

Here's a Wireshark trace that shows the join:

![image](https://user-images.githubusercontent.com/64669/71124612-3b8a3f80-21b3-11ea-8af9-84bbbdf922db.png)

The weird part is that it leaves the group about 10 seconds after it joins it. That means that `nerves.local` works for a short time and then stops working. My computer caches `nerves.local` so that means that sometimes it gets the address for real and sometimes from cache. That makes it a little confusing to know why it works sometimes and not other times.

I added a tool called `multicast_addresses` to `toolshed`. It's in a branch now, but I plan to clean it up a little more and merge it down. It lists out all of the multicast addresses that are currently being listened on. If you want to reproduce this, you can poll it and watch the `224.0.0.251` appear and disappear.

This is very repeatable. I'm currently using an x86_64-based board and `nerves_pack`. `mdns_lite` looks like it's running fine and using `dig` to resolve `nerves.local` via unicast works fine.

I believe that the loss of the multicast registration is correlated with `udhcpc` being restarted in the logs:

```
21:22:55.014 [info]  r8169 0000:01:00.0 eth0: link down

21:22:55.015 [info]  IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready

21:22:55.015 [info]  r8169 0000:01:00.0 eth0: link down

21:22:55.024 [debug] /sbin/udhcpc: udhcpc: started, v1.31.1

21:22:55.047 [debug] /sbin/udhcpc: udhcpc: sending discover

21:22:58.091 [debug] /sbin/udhcpc: udhcpc: sending discover

21:23:01.123 [debug] /sbin/udhcpc: udhcpc: sending discover

21:23:07.377 [info]  r8169 0000:01:00.0 eth0: link up

21:23:07.377 [info]  IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready

21:23:24.499 [debug] /sbin/udhcpc: udhcpc: sending discover

21:23:25.514 [debug] /sbin/udhcpc: udhcpc: sending select for 192.168.9.247

21:23:25.834 [debug] /sbin/udhcpc: udhcpc: lease of 192.168.9.247 obtained, lease time 86400

21:23:25.838 [debug] udhcpc.renew(eth0): %{dns: ["192.168.9.1"], domain: "hunleth.lan", interface: "eth0", ip: "192.168.9.247", lease: "86400", mask: "24", opt53: "05", router: ["192.168.9.1"], serverid: "192.168.9.1", subnet: "255.255.255.0"}

21:23:25.844 [info]  RouteManager: set_route eth0 -> :lan

21:23:38.935 [debug] /sbin/udhcpc: udhcpc: started, v1.31.1

21:23:38.937 [info]  RouteManager: clear_route eth0

21:23:38.964 [debug] /sbin/udhcpc: udhcpc: sending discover

21:23:39.000 [debug] /sbin/udhcpc: udhcpc: sending select for 192.168.9.247

21:23:39.323 [debug] /sbin/udhcpc: udhcpc: lease of 192.168.9.247 obtained, lease time 86400

21:23:39.327 [debug] udhcpc.renew(eth0): %{dns: ["192.168.9.1"], domain: "hunleth.lan", interface: "eth0", ip: "192.168.9.247", lease: "86400", mask: "24", opt53: "05", router: ["192.168.9.1"], serverid: "192.168.9.1", subnet: "255.255.255.0"}
```

**Comment by fhunleth** [comment]
I'm pretty sure that this isn't an `mdns_lite` issue. Sorry for the noise.

**Comment by fhunleth** [comment]
Update: the issue that caused this was fixed in `vintage_net v0.7.1`.
